packages/smooth_app/lib/pages/product/edit_ocr_page.dart

```diff
@@ -377,4 +375,224 @@ class _EditOcrPageState extends State<EditOcrPage> with UpToDateMixin {
+                  fontSize: 15.5,
+                  color: Colors.white,
+                ),
+                child: child,
+              ),
+            ),
+          ),
+        ),
+      ),
+    );
+  }
+
+  double _computeFontScaleFactor(BuildContext context) {
+    final double fontSize = DefaultTextStyle.of(context).style.fontSize ?? 15.0;
+    final double scaledFontSize =
+        MediaQuery.textScalerOf(context).scale(fontSize);
+
+    return scaledFontSize / fontSize;
+  }
+}
+
+class _EditOcrActionExtractingContent extends StatelessWidget {
+  const _EditOcrActionExtractingContent({
+    required this.helper,
+    required this.appLocalizations,
+  });
+
+  final OcrHelper helper;
+  final AppLocalizations appLocalizations;
+
+  @override
+  Widget build(BuildContext context) {
+    return Semantics(
+      label: helper.getActionLoadingPhoto(appLocalizations),
+      excludeSemantics: true,
+      child: Shimmer(
+        gradient: const LinearGradient(
+          begin: Alignment.centerLeft,
+          end: Alignment.centerRight,
+          colors: <Color>[
+            Colors.black,
+            Colors.white,
+            Colors.black,
+          ],
+        ),
+        child: Padding(
+          padding: const EdgeInsets.symmetric(horizontal: MEDIUM_SPACE),
+          child: Row(
+            mainAxisAlignment: MainAxisAlignment.center,
+            crossAxisAlignment: CrossAxisAlignment.center,
+            children: <Widget>[
+              const CircularProgressIndicator.adaptive(),
+              Expanded(
+                child: Text(
+                  helper.getActionExtractingData(appLocalizations),
+                  textAlign: TextAlign.center,
+                ),
+              ),
+              const CircularProgressIndicator.adaptive(),
+            ],
+          ),
+        ),
+      ),
+    );
+  }
+}
+
+class _ExtractMainActionContentLoaded extends StatelessWidget {
+  const _ExtractMainActionContentLoaded({
+    required this.helper,
+    required this.appLocalizations,
+    required this.onPressed,
+  });
+
+  final OcrHelper helper;
+  final AppLocalizations appLocalizations;
+  final VoidCallback onPressed;
+
+  @override
+  Widget build(BuildContext context) {
+    return Semantics(
+      excludeSemantics: true,
+      value: helper.getActionExtractText(appLocalizations),
+      button: true,
+      child: InkWell(
+        onTap: onPressed,
+        borderRadius: ANGULAR_BORDER_RADIUS,
+        child: Padding(
+          padding: const EdgeInsets.symmetric(horizontal: MEDIUM_SPACE),
+          child: Row(
+            mainAxisAlignment: MainAxisAlignment.center,
+            crossAxisAlignment: CrossAxisAlignment.center,
+            children: <Widget>[
+              Expanded(
+                child: Text(
+                  helper.getActionExtractText(appLocalizations),
+                ),
+              ),
+              const Icon(
+                Icons.download,
+                semanticLabel: '',
```

Comment:
Is that worth it?

Reply:
Yes, otherwise, the screen reader tries to say the character in font (that's the default behavior)

Reply:
> Yes, otherwise, the screen reader tries to say the character in font (that's the default behavior)

Hmm, I guess that wouldn't hurt to say "download".
If I've understood well, that also means that we're supposed to give each `Icon` we use in the app that meaningless `semanticLabel: ''` parameter, which is not nice.

Reply:
That's not needed, because we have the text just before.